### DISCUSSION TITLE
General UI

### DISCUSSION THREAD

**Discussion by hmltn2** [mozilla-mobile/firefox-ios#13368] [General]
Hey,

I have spent more time recently going through some of my applications very thoroughly to understand thoroughly.

After going more thoroughly through Firefox and Safari on iOS I ended on feeling like the UI for Firefox is actually pretty lacking, unfortunately.

Even if Firefox generally is a browser with good functionalities, especially on desktop, I’m not sure but it feels like there was less priority put on the iOS app.

There are just a lot of slightly cumbersome interfaces or menus, and Safari just feels so much more usable, from a touch interface.

I’d like to try to go deeper and contribute to Firefox, so I’ll check out some of the open issues. That said, has anyone considered the UI could use a revamp?

If I feel like pursuing this I suppose I could fork it and explore a bit if I can work on any changes I might think could be nice.

In that case, is there some kind of protocol which any version of Firefox has to conform to? Like, who makes the decisions about what features Firefox will and will not have? Who do the developers report to, in decided if they want to add a feature or take it away?

Thanks.

**Comment by lmarceau** [comment]
Hey @hmltn2,
We always aim to improve, if you think of any ways to upgrade the application, please let us know by writing a [feature request ](https://github.com/mozilla-mobile/firefox-ios/issues/new?assignees=&labels=Feature-Request&template=---feature.md) ticket. Tickets gets triaged by the product team. We also have a UX team that will look into proposals. If you want to implement changes for a certain feature request, before doing the work, you should ask on the ticket to get a confirmation said work could land into the app (wouldn't want you to open a PR that wouldn't land). Not all feature requests by users get enabled. We also have some [contributor friendly tickets](https://github.com/mozilla-mobile/firefox-ios/labels/Contributor%20OK) if you are interested in getting started with contributions.
I hope this helps, thank you!
